tmk_core/common/keycode.h

```diff
@@ -519,7 +520,14 @@ enum internal_special_keycodes {
     KC_FN28,
     KC_FN29,
     KC_FN30,
-    KC_FN31
+    KC_FN31,
+
+    KC_PROGRAMMABLE_BUTTON0 = 0xE8,
```

Comment:
There is no “Button 0” in the HID Usage Tables document — 0 is reserved for “No Button Pressed”, and the button numbering starts from 1. So you probably need to make the first enum value for these buttons `KC_PROGRAMMABLE_BUTTON1`, which is actually similar to the mouse button keycodes (`KC_MS_BTN1`).

Reply:
Good point, will do.